engines/dfc_provider/app/controllers/dfc_provider/api/base_controller.rb

```diff
@@ -26,17 +25,17 @@ def check_user
         head :unauthorized
       end
 
-      def set_enterprise
-        current_enterprise
+      def check_enterprise
+        return if current_enterprise.present?
+
+        not_found
```

Comment:
I think that `find` in `current_enterprise` raises an error if the enterprise can't be found. And that will lead to a NOT FOUND anyway. So I would guess that `check_enterprise` is still not needed and you could just call `current_enterprise` instead. This code makes the intent clearer though.

Reply:
Right, but I thought it would have been more explicit to do this check directly. Like for the `user` for example.

Also, we have a design issue the data model for DFC. I was having an issue inside the `CatalogItemsController` where the `current_enterprise` is not directly called. I will call it manually there, with a comment explaining the issue.